src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -461,6 +531,15 @@ const mapStateToProps = (state, ownProps) => ({
         sGetUiConditionsByDimension(state, ownProps.dimension?.id) || {},
     dimensionIdsInLayout: sGetDimensionIdsFromLayout(state),
     displayNameProp: sGetSettingsDisplayNameProperty(state),
+    inputType: sGetUiInputType(state),
+    stage:
+        sGetMetadataById(
+            state,
+            ownProps.dimension?.id?.substring(
```

Comment:
Could it be that `dimension.id` does not contain the program stage id?
In that case this code won't work.

Reply:
In the case the dimension doesn't have a stage prefix (i.e. there is no stage) then `{}` is returned and there will be no `stage` for this dimension...